client/app/routes/collection/add.js

```diff
@@ -1,4 +1,124 @@
+                'submit'
+            ],
+            initial_state: {
+                upload_section: {
+                    state: ['unsaved', 'editing'],
+                    value: undefined
+                },
+                disciplines_section: {
+                    state: ['disabled'],
+                    value: undefined
+                },
+                basic_info_section: {
+                    state: ['disabled'],
+                    value: undefined
+                },
+                authors_section: {
+                    state:  ['disabled'],
+                    value: undefined
+                },
+                submit_button: {
+                    state: ['disabled'],
+                    value: undefined
+                },
+                preprint_file_upload_widget: {
+                    state: ['undefined'],
+                    value: undefined
+                },
+                preprint_title_widget: {
+                    state: ['undefined'],
+                    value: undefined
+                },
+                save_upload_section_widget: {
+                    state: ['undefined'],
+                    value: undefined
+                }
+            },
+            actions: [{
+                type: 'create_widget',
+                parameters: {
+                    widget_component: 'file-uploader',
+                    description: 'Choose the preprint file to upload',
+                    section: 'upload',
+                    output: 'preprint_file_data'
+                },
+                output: 'preprint_file_upload_widget',
+                conditions: [{
+                    all: [{
+                        parameter: 'upload_section',
+                        state: 'unsaved',
+                    }, {
+                        parameter: 'preprint_file_upload_widget',
+                        state: 'undefined'
+                    }]
+                }]
+            }, {
+                type: 'create_widget',
+                parameters: {
+                    widget_component: 'text-field',
+                    description: 'Enter the title for this preprint',
+                    section: 'upload',
+                    output: 'preprint_file_name'
+                },
+                output: 'preprint_title_widget',
+                conditions: [{
+                    all: [{
+                        parameter: 'preprint_title_widget',
+                        state: 'undefined',
+                    }, {
+                        parameter: 'preprint_file_data',
+                        state: 'defined',
+                    }, {
+                        parameter: 'upload_section',
+                        state: 'editing',
+                    }],
+                }]
+            }, {
+                type: 'create_widget',
+                parameters: {
+                    widget_component: 'button-widget',
+                    description: 'Save this section',
```

Comment:
Since the upload section can be saved only when all of its components have a value, I think this can be simplified. Perhaps with a computed property `isValid` that checks for values of all of the widgets with `section` "upload".